packages/react/src/primitives/shared/utils.ts

```diff
@@ -27,18 +22,11 @@ export const getConsecutiveIntArray = (start: number, end: number) => {
 
 /**
  * Create a uuid to use with amplify fields unless
- * an id is already defined
+ * an id is provided
  * @param id user specified id or undefined
  * @returns string
  */
-export const useAmplifyFieldID = (id?: string) => {
-  return React.useMemo(() => {
-    if (id) {
-      return id;
-    }
-    return `amplify-field-${nanoid()}`;
-  }, [id]);
-};
+export const useAmplifyFieldID = (id?: string) => useId(id);
```

Comment:
nit: I was wondering if this function only specific to field. Should we make it more generic like `useAmplifyID` so that it could be used to generated random id for other primitives as well?

Reply:
That's a good idea, I'll update.